File: query_results_no_authentication.py
import requests, json
import pandas as pd

# make empty list to store results

total_results = []

# make list of page numbers (adapt range to expected n of results)

for page_num in range(1, 40):
    url = "https://api.druid.datalegend.net/queries/RubenS/all-obs/3/run?pageSize=10000&page=" + str(page_num)
    print(url)

# Loop through from pages (adapt range to expected n of results)

for page_num in range(1, 40):
    # Build the URL and download the results
    url = "https://api.druid.datalegend.net/queries/RubenS/all-obs/3/run?pageSize=10000&page=" + str(page_num)
    print("Downloading", url)
    response = requests.get(url)
    data = response.json()
    total_results = total_results + data

# write total_results to csv

df = pd.DataFrame(total_results)
#data.read_json(data)
df.to_csv('results.csv')

File: query_results_no_authentication.R
data <- read.csv(url("https://api.druid.datalegend.net/queries/LaurensRietveld/pagination/run.csv?page=1&pageSize=10000"))
page = 1 # counts how many pages you queried so far
cont = TRUE # tells the loop to continue
while(cont && page < 100){ # prevents an endless loop; adjust the maximum number of pages based on your dataset's size.
  page = page+1
  n1 = nrow(data) # how many rows existed before
  data <- rbind(data, read.csv(url(sprintf("https://api.druid.datalegend.net/queries/LaurensRietveld/pagination/run.csv?page=%i&pageSize=10000", page)))) # queries the next page
  n2 = nrow(data) # how many rows exist after
  cont = (n2 - n1) == 10000 # checks whether the next query will pick any more rows; if not, the loop stops
  #print(page) # if you want to follow what's going on
}

### it is possible that in some instances not all results are read correctly by R. Perhaps due to read.csv or the specification .csv in the api? 
### can be solved by using fromJSON and removing the .csv specification from the url:

ibrary(jsonlite)
library(rjson)

data <- jsonlite::fromJSON("https://api.druid.datalegend.net/queries/dataLegend/microheights-to-table/15/run?page=1&pageSize=10000")
page = 1 # counts how many pages you queried so far
cont = TRUE # tells the loop to continue
while(cont && page < 1000){ # prevents an endless loop; adjust the maximum number of pages based on your dataset's size.
  page = page+1
  n1 = nrow(data) # how many rows existed before
  data <- rbind(data, jsonlite::fromJSON(sprintf("https://api.druid.datalegend.net/queries/dataLegend/microheights-to-table/15/run?page=%i&pageSize=10000", page))) # queries the next page
  n2 = nrow(data) # how many rows exist after
  cont = (n2 - n1) == 10000 # checks whether the next query will pick any more rows; if not, the loop stops
  print(page) # if you want to follow what's going on
}
